File: RepoTransBench/target_projects/Java/Go/googlearchive_android-ClippingBasic/tests/original/messageonlylogfilter_test.go
package original

import (
	"errors"
	"testing"

	"clippingbasic/tests"
)

type RecordingNode struct {
	LastPriority int
	LastTag      string
	LastMsg      string
	LastTr       error
}

func (n *RecordingNode) Println(priority int, tag, msg string, tr error) {
	n.LastPriority = priority
	n.LastTag = tag
	n.LastMsg = msg
	n.LastTr = tr
}

func setupFilter() (*tests.MessageOnlyLogFilter, *RecordingNode) {
	recorder := &RecordingNode{LastPriority: -100, LastTag: "zzz", LastMsg: "uuu"}
	filter := &tests.MessageOnlyLogFilter{}
	filter.SetNext(recorder)
	return filter, recorder
}

func TestPrintlnFiltersToMessageOnly(t *testing.T) {
	filter, recorder := setupFilter()
	filter.Println(tests.INFO, "TAG", "hellomsg", errors.New("should not propagate"))
	if recorder.LastPriority != tests.NONE {
		t.Errorf("Expected priority NONE, got %d", recorder.LastPriority)
	}
	if recorder.LastTag != "" {
		t.Errorf("Expected empty tag, got '%s'", recorder.LastTag)
	}
	if recorder.LastMsg != "hellomsg" {
		t.Errorf("Expected message 'hellomsg', got '%s'", recorder.LastMsg)
	}
	if recorder.LastTr != nil {
		t.Errorf("Expected nil error")
	}
}

func TestConstructorWithNext(t *testing.T) {
	recorder := &RecordingNode{}
	filter := tests.NewMessageOnlyLogFilter(recorder)
	if filter.GetNext() != recorder {
		t.Errorf("Expected Next to match recorder")
	}
}

func TestSetAndGetNext(t *testing.T) {
	recorder := &RecordingNode{}
	f := &tests.MessageOnlyLogFilter{}
	f.SetNext(recorder)
	if f.GetNext() != recorder {
		t.Errorf("Expected Next to match recorder on set")
	}
}

func TestNullNextDoesNothing(t *testing.T) {
	f := &tests.MessageOnlyLogFilter{}
	defer func() {
		if r := recover(); r != nil {
			t.Error("Should not panic when Next is nil")
		}
	}()
	f.Println(tests.ERROR, "x", "yz", nil)
}

File: RepoTransBench/target_projects/Java/C#/googlearchive_android-ClippingBasic/tests/original/MessageOnlyLogFilterTest.cs
using System;
using Xunit;
using ClippingBasicSample.Log;

namespace ClippingBasicSample.Tests.Original
{
    public class MessageOnlyLogFilterTest
    {
        private class RecordingNode : ILogNode
        {
            public int LastPriority = int.MinValue;
            public string LastTag = "zzz";
            public string LastMsg = "uuu";
            public Exception LastTr;

            public void Println(int priority, string tag, string msg, Exception tr)
            {
                LastPriority = priority;
                LastTag = tag;
                LastMsg = msg;
                LastTr = tr;
            }
        }

        private MessageOnlyLogFilter _filter;
        private RecordingNode _recorder;

        public MessageOnlyLogFilterTest()
        {
            _recorder = new RecordingNode();
            _filter = new MessageOnlyLogFilter();
            _filter.SetNext(_recorder);
        }

        [Fact]
        public void TestPrintlnFiltersToMessageOnly()
        {
            _filter.Println(Log.INFO, "TAG", "hellomsg", new Exception("should not propagate"));
            Assert.Equal(Log.NONE, _recorder.LastPriority);
            Assert.Null(_recorder.LastTag);
            Assert.Equal("hellomsg", _recorder.LastMsg);
            Assert.Null(_recorder.LastTr);
        }

        [Fact]
        public void TestConstructorWithNext()
        {
            var f2 = new MessageOnlyLogFilter(_recorder);
            Assert.Equal(_recorder, f2.GetNext());
        }

        [Fact]
        public void TestSetAndGetNext()
        {
            var f3 = new MessageOnlyLogFilter();
            f3.SetNext(_recorder);
            Assert.Equal(_recorder, f3.GetNext());
        }

        [Fact]
        public void TestNullNextDoesNothing()
        {
            var f4 = new MessageOnlyLogFilter();
            // Should not throw
            f4.Println(Log.ERROR, "x", "yz", null);
        }
    }
}